src/integration_tests/rpc_multiple_remote_addrs/main.cc

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2016 Alexander Gallego. All rights reserved.
+//
+// std
+#include <chrono>
+#include <iostream>
+// linux
+#include <arpa/inet.h>
+#include <ifaddrs.h>
+#include <net/if.h>
+// seastar
+#include <core/app-template.hh>
+#include <core/distributed.hh>
+#include <core/sleep.hh>
+#include <net/api.hh>
+// smf
+#include "integration_tests/non_root_port.h"
+#include "smf/log.h"
+#include "smf/random.h"
+#include "smf/rpc_generated.h"
+#include "smf/rpc_handle_router.h"
+#include "smf/rpc_recv_context.h"
+#include "smf/rpc_server.h"
+// templates
+#include "integration_tests/demo_service.smf.fb.h"
+
+struct client_info {
+    explicit client_info(seastar::ipv4_addr ip): client(smf_gen::demo::SmfStorageClient::make_shared(ip)), ip(ip.ip) {};
+    client_info(): client(nullptr), ip(0), ip_str("") {};
```

Comment:
ip_str is a bug. 

You are assigning a global constant string literal `""` to the char array ip_str - you've already zero it out by doing `{}` no need. 

also you already field initialize all your member fields.  so you just have to do 

`client_info(){}` that's it.